contracts/kleros/KlerosLiquid.sol

```diff
@@ -0,0 +1,724 @@
+        courts[_subcourtID].timesPerPeriod = _timesPerPeriod;
+    }
+
+    /** @dev Pass the phase. */
+    function passPhase() external {
+        if (phase == Phase.staking) {
+            // solium-disable-next-line security/no-block-members
+            require(block.timestamp - lastPhaseChange >= minStakingTime, "The minimum staking time has not passed yet.");
+            require(disputesWithoutJurors > 0, "There are no disputes without jurors.");
+            RNBlock = block.number + 1;
+            _RNG.requestRN(RNBlock);
+            phase = Phase.generating;
+        } else if (phase == Phase.generating) {
+            RN = _RNG.getUncorrelatedRN(RNBlock);
+            require(RN != 0, "Random number is not ready yet.");
+            phase = Phase.drawing;
+        } else if (phase == Phase.drawing) {
+            // solium-disable-next-line security/no-block-members
+            require(disputesWithoutJurors == 0 || block.timestamp - lastPhaseChange >= maxDrawingTime, "There are still disputes without jurors and the maximum drawing time has not passed yet.");
+            phase = Phase.staking;
+        }
+
+        // solium-disable-next-line security/no-block-members
+        lastPhaseChange = block.timestamp;
+        emit NewPhase(phase);
+    }
+
+    /** @dev Pass the period of a specified dispute.
+     *  @param _disputeID The ID of the dispute.
+     */
+    function passPeriod(uint _disputeID) external {
+        Dispute storage dispute = disputes[_disputeID];
+        if (dispute.period == Period.evidence) {
+            // solium-disable-next-line security/no-block-members
+            require(block.timestamp - dispute.lastPeriodChange >= courts[dispute.subcourtID].timesPerPeriod[uint(dispute.period)], "The evidence period time has not passed yet.");
+            require(dispute.appealDraws[dispute.appealDraws.length - 1] == dispute.votes[dispute.votes.length - 1].length, "The dispute has not finished drawing yet.");
+            dispute.period = courts[dispute.subcourtID].hidden ? Period.commit : Period.vote;
+        } else if (dispute.period == Period.commit) {
+            require(
+                // solium-disable-next-line security/no-block-members
+                block.timestamp - dispute.lastPeriodChange >= courts[dispute.subcourtID].timesPerPeriod[uint(dispute.period)] || dispute.appealCommits[dispute.appealCommits.length - 1] == dispute.votes[dispute.votes.length - 1].length,
+                "The commit period time has not passed yet and not every juror has committed yet."
+            );
+            dispute.period = Period.vote;
+        } else if (dispute.period == Period.vote) {
+            require(
+                // solium-disable-next-line security/no-block-members
+                block.timestamp - dispute.lastPeriodChange >= courts[dispute.subcourtID].timesPerPeriod[uint(dispute.period)] || dispute.appealVotes[dispute.appealVotes.length - 1] == dispute.votes[dispute.votes.length - 1].length,
```

Comment:
Only check time

Reply:
That's why there is a `||`/`or` operator. You can pass the period before the time passes if everyone has committed/voted.